src/main/java/com/atlassian/bitbucket/jenkins/internal/applink/oauth/serviceprovider/consumer/Consumer.java

```diff
@@ -169,7 +169,7 @@ public Builder(String key) {
          * @return {@code this} builder
          */
         public Builder name(String name) {
-            this.name = checkNotNull(name);
+            this.name = requireNonNull(name);
```

Comment:
Very high level: I'd use it for non-optional fields (which I usually pass in the builder's constructor), but not for the rest, because you can have the following (very) hypothetical code:
```
// Builder copies all the values form existing instance on initialization
Builder builder = new Builder(existingInstanceIWannaCopy);
...
if (some condition) {
   // if some condition is true, I wanna un-set a certain field in the new instance I'm building
   builder.withX(null);
}
...
```
In this case if `withX()` has a `requireNonNull()`, obviously I'd be in trouble.
I know this example is very specific, but as a general principle, I try not to use `requireNonNull()` in builders, and instead a) pass non-optional values in the builder constructor, _with_ a `requireNonNull()`, and b) for other non-optional stuff _not_ passed into the constructor (e.g. for conditionally optional stuff, like "if X set then Y must also be set"), I check in the `build()` method.